src/main/java/frc/robot/subsystems/Grabber.java

```diff
@@ -0,0 +1,123 @@
+    if(armPositions.length>=2)
+      setArmPosition(armPositions);
+    this.armMode = armMode;
+  }
+
+  public void setEndEffectorMode(EFModes effectorMode){
+    switch(effectorMode){
+      case INTAKING_CUBE: endEffectorSubsystem.intakeCube(); break;
+      case INTAKING_CONE: endEffectorSubsystem.intakeCone(); break;
+      case HOLDING_CONE: endEffectorSubsystem.holdCone(); break;
+      case HOLDING_CUBE: endEffectorSubsystem.holdCube(); break;
+      case PLACING_CONE: endEffectorSubsystem.placeCone(); break;
+      case PLACING_CUBE: endEffectorSubsystem.placeCube(); break;
+      case STOPPED: endEffectorSubsystem.stopMotors(); break;
+    }
+    efMode = effectorMode;
+  }
+
+  public double[] getArmPositions(ArmModes armMode){
+    switch (armMode){
+      case INTAKING_CUBE: return ArmConstants.intakingCubesArmPos;
+      case INTAKING_CONE_UPRIGHT: return ArmConstants.intakingConesUprightArmPos;
+      case INTAKING_CONE_FALLEN: return ArmConstants.intakingConesFallenArmPos;
+      case HOLDING: return ArmConstants.holdingArmPos;
+
+      case PLACING_CONE_LVL1: return ArmConstants.placingConeArmPosOne;
+      case PLACING_CONE_LVL2: return ArmConstants.placingConeArmPosTwo;
+      case PLACING_CONE_LVL3: return ArmConstants.placingConeArmPosThree;
+
+      case PLACING_CUBE_LVL1: return ArmConstants.placingCubeArmPosOne;
+      case PLACING_CUBE_LVL2: return ArmConstants.placingCubeArmPosTwo;
+      case PLACING_CUBE_LVL3: return ArmConstants.placingCubeArmPosThree;
+    }
+    return new double[]{};
+  }
+
+  private EFModes efMode = EFModes.STOPPED;
+  private ArmModes armMode = ArmModes.HOLDING;
+  private EFModes desiredEFMode = EFModes.STOPPED;
+  private ArmModes desiredArmMode = ArmModes.HOLDING;
+
+  public void setDesiredEFMode(EFModes desiredEFMode){
+    this.desiredEFMode = desiredEFMode;
+  }
+  public void setDesiredArmMode(ArmModes desiredArmMode){
+    this.desiredArmMode = desiredArmMode;
+  }
+  public EFModes getDesiredEFMode(){
+    return desiredEFMode;
+  }
+  public ArmModes getDesiredArmMode(){
+    return desiredArmMode;
+  }
+
+  public EFModes getEFMode(){
+    return efMode;
+  }
+  public ArmModes getArmMode(){return armMode;}
+
+
+
+
+  public void setArmPosition(double[] armPosition){
+    if(armPosition.length<2) return;
+    armStageOne.setArmPositionRad(armPosition[0]);
+    armStageOne.setArmPositionRad(armPosition[1]);
+  }
+
+  public void setDesiredArmAndEFModes(ArmModes armMode, EFModes efMode){
+    setDesiredEFMode(efMode);
+    setDesiredArmMode(armMode);
+  }
+
+
+  @Override
+  public void periodic() {
+    // This method will be called once per scheduler run
+
+    if(getEFMode() != getDesiredEFMode()){ //wait for the arm to be within the tolerance to update the EF mode if intaking or placing
+      boolean armOneAligned = MathThings.isWithinRangeOf(armStageOne.getEncoderRad(), getArmPositions(getDesiredArmMode())[0],ArmConstants.angleToleranceToUpdateEF);
```

Comment:
maybe rename maththings to MBLibMath? (MB is mechanical Bulls and we can start all library files with that? I also think maybe util and lib should be separate if there are util things that we want out of our subsystems, but not in a persistent library)